### PR TITLE
ref(crons): Include message_type in kafka message

### PR DESCRIPTION
**Pull request by evanpurkhiser**
There are two types of messages that end up in the ingest-monitors kafka topic, "check_in" (the ones produced here in relay) and "clock_pulse" messages, which are produced externally and are intended to ensure the clock continues to run even when ingestion volume drops.

Currently we have some shim logic in our producer which handles adding in this message type when it is missing

https://github.com/getsentry/sentry/blob/07522eb8c43828deee6a74de2bb72cf989ec907d/src/sentry/monitors/consumers/monitor_consumer.py#L591-L595

This change introduces the `message_type` key for the check-in messages so we no longer need to shim this.

#skip-changelog

### CODE REVIEW

File: relay-server/src/actors/store.rs [hunk 1]
```diff
@@ -1028,6 +1029,9 @@ struct CheckInKafkaMessage {
     #[serde(skip)]
     routing_key_hint: Option<Uuid>,
 
+    /// Discriminate tag used to differentiate these messages from clock pulse messages that are
+    /// also placed into the topic.
+    message_type: String,
```
Comment (Dav1dde) [inline relay-server/src/actors/store.rs]: Should we use an enum for this?

Comment (evanpurkhiser) [reply relay-server/src/actors/store.rs]: Oh yeah, I like that idea. 👍

File: relay-server/src/actors/store.rs [hunk 2]
```diff
@@ -1028,6 +1029,9 @@ struct CheckInKafkaMessage {
     #[serde(skip)]
     routing_key_hint: Option<Uuid>,
 
+    /// Discriminate tag used to differentiate these messages from clock pulse messages that are
+    /// also placed into the topic.
```
Comment (iker-barriocanal) [inline relay-server/src/actors/store.rs]: Could we reword this to meet [relay coding guidelines (internal)](https://www.notion.so/sentry/Relay-Code-and-Architecture-4fac481cf27741aba54d39594f1cae58?pvs=4#987c0416a9f54674999eed59eecbf846)? Thanks!

> ensure that there is a precise first line, followed by more information

**Review by iker-barriocanal — APPROVED** [review]

File: relay-server/src/actors/store.rs (hunk 2 above)
Comment (evanpurkhiser) [reply relay-server/src/actors/store.rs]: I think I fixed this

**Comment by evanpurkhiser** [comment]
Gonna get one more look over this just to be safe.

File: relay-server/src/actors/store.rs [hunk 3]
```diff
@@ -1023,11 +1024,21 @@ struct ProfileKafkaMessage {
     payload: Bytes,
 }
 
+#[allow(dead_code)]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "snake_case")]
+enum CheckInMessageType {
```
Comment (Dav1dde) [inline relay-server/src/actors/store.rs:1030]: The first sentence of your issue would be nice here. Documenting the use of the values and that relay is only supposed to ever send `CheckIn`'s

**Review by Dav1dde — APPROVED** [review]

**Review by iker-barriocanal — APPROVED** [review]
🚀

File: relay-server/src/actors/store.rs (hunk 3 above)
Comment (evanpurkhiser) [reply relay-server/src/actors/store.rs:1030]: I'll add it in in a diff PR 👍
